### ISSUE TITLE
🐛 <TITLE>

### ISSUE THREAD

**Issue by Aeolun** [biomejs/biome#932]
### Environment information

```block
❯ node_modules/.bin/biome rage
CLI:
  Version:                      1.4.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-kitty"
  JS_RUNTIME_VERSION:           "v18.18.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```
```


### What happened?

```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_service/src/configuration/linter/mod.rs:167:18
Thread Name: biome::worker_7
Message: internal error: entered unreachable code: the rule is turned off, it should not step in here
```

### Expected result

Biome does not ask me to report an issue.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Please create a reproduction, so we can easily help you

**Comment by believer** [comment]
This might be a duplicate of #691?

**Comment by ematipico** [comment]
Probably, but still, I would like to have a repository to test. The issue is open for a week without updates from OP, I close it.

**Comment by Aeolun** [comment]
Respectfully, your client asks me to report it. Not to do any extra work to create a reproduction. Creating an issue for a problem I do not care about is low effort for me. Creating a repository to reproduce the issue is not. 

Maybe it should say: “We might not do anything with it unless you also create a repository with a reproduction.”?

I realize your time is valuable, so I’m totally on board with that. But so is mine, so it’s nice to know how useful just ‘reporting’ the issue is. I might refrain from doing so until it’s valuable enough to me I have incentive to create a reproduction.
